golang/cosmos/x/swingset/types/msgs.go

```diff
@@ -27,16 +30,14 @@ var (
 
 // Charge an account address for the beans associated with given messages and storage.
 // See list of bean charges in default-params.go
-func chargeAdmission(ctx sdk.Context, keeper SwingSetKeeper, addr sdk.AccAddress, msgs []string, storage []string) error {
+func chargeAdmission(ctx sdk.Context, keeper SwingSetKeeper, addr sdk.AccAddress, msgs []string, storageLen uint64) error {
 	beansPerUnit := keeper.GetBeansPerUnit(ctx)
 	beans := beansPerUnit[BeansPerInboundTx]
 	beans = beans.Add(beansPerUnit[BeansPerMessage].Mul(sdk.NewUint(uint64(len(msgs)))))
 	for _, msg := range msgs {
 		beans = beans.Add(beansPerUnit[BeansPerMessageByte].Mul(sdk.NewUint(uint64(len(msg)))))
 	}
-	for _, store := range storage {
-		beans = beans.Add(beansPerUnit[BeansPerStorageByte].Mul(sdk.NewUint(uint64(len(store)))))
-	}
+	beans = beans.Add(beansPerUnit[BeansPerStorageByte].MulUint64(storageLen))
```

Comment:
If `.MulUint64` works as it says on the tin, maybe replace the other uses of `.Mul(sdk.NewUint(uint64(` above.

Reply:
Done.